simphony/io/tests/test_file_lattice.py

```diff
@@ -0,0 +1,180 @@
+        CustomRecord column description has correct attributes
+
+        """
+        self.assertEqual(self.filelattice.name, 'test_lattice')
+        self.assertEqual(self.filelattice.type, 'cubic')
+        self.assertItemsEqual(self.filelattice.base_vect,
+                              numpy.array((1.0, 1.0, 1.0)))
+        self.assertItemsEqual(self.filelattice.size, (4, 5, 6))
+        self.assertItemsEqual(self.filelattice.origin,
+                              numpy.array((0.0, 0.0, 0.0)))
+
+    def test_file_lattice_constructed_from_table_existing_in_file(self):
+        """ Checks that FileLattice constructed from a known table already
+        existing in H5CUDS-file provides correct attributes for FileLattice
+        object
+
+        """
+        self.filelattice = self.file.get_lattice('test_lattice')
+
+        self.assertEqual(self.filelattice.name, 'test_lattice')
+        self.assertEqual(self.filelattice.type, 'cubic')
+        self.assertItemsEqual(self.filelattice.base_vect,
+                              numpy.array((1.0, 1.0, 1.0)))
+        self.assertItemsEqual(self.filelattice.size, (4, 5, 6))
+        self.assertItemsEqual(self.filelattice.origin,
+                              numpy.array((0.0, 0.0, 0.0)))
+
+    def test_get_node(self):
+        """ Check that a LatticeNode can be retrieved correctly
+
+        """
+        N = self.filelattice.get_node((3, 3, 3))
+
+        self.assertTrue(isinstance(N, LatticeNode))
+        self.assertItemsEqual(N.index, (3, 3, 3))
+
+        for key, value in self.lattice.get_node((3, 3, 3)).data.iteritems():
+            self.assertEqual(N.data[key], value)
+
+    def test_node_iterator(self):
+        """ Checks the node iterator
+
+        Checks that an iterator over all
+        the nodes is returned when the function iter_nodes is called
+        without arguments
+
+        """
+        fl_nodes = self.filelattice.iter_nodes()
+
+        for M in fl_nodes:
+            N = self.lattice.get_node(M.index)
+            self.assertEqual(N.index, M.index)
+            self.assertItemsEqual(N.data, M.data)
+
+    def test_node_iterator_subset(self):
+        """ Checks the node iterator on a subset of nodes
+
+        """
+
+        fl_nodes = self.filelattice.iter_nodes([(0, 0, 0), (0, 1, 2)])
+
+        for M in fl_nodes:
+            N = self.lattice.get_node(M.index)
+            self.assertEqual(N.index, M.index)
+            self.assertItemsEqual(N.data, M.data)
+
+    def test_update_node(self):
+        """ Check that a node can be updated correctly
+
+        """
+
+        N = LatticeNode((3, 2, 3),
+                        {CUBA.MATERIAL_ID: 2, CUBA.DENSITY: 10.0,
+                         CUBA.VELOCITY: (0.0, 0.0, 10.0)})
+
+        self.filelattice.update_node(N)
+
+        M = self.filelattice.get_node((3, 2, 3))
+
+        self.assertItemsEqual(N.index, M.index)
+
+        for key, value in M.data.iteritems():
```

Comment:
see comment See comment in https://github.com/simphony/simphony-common/pull/74/files#r25767404